react-components/src/components/Reveal3DResources/useCadOrPointCloudResources.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { type AddModelOptions, type ClassicDataSourceType } from '@cognite/reveal';
+import { useMemo } from 'react';
+import { useModelIdRevisionIdFromModelOptions } from '../../hooks/useModelIdRevisionIdFromModelOptions';
+import { isDefined } from '../../utilities/isDefined';
+import { is360ImageAddOptions } from './typeGuards';
+import {
+  type AddResourceOptions,
+  type AddCadResourceOptions,
+  type AddPointCloudResourceOptions
+} from './types';
+
+export const useCadOrPointCloudResources = (
+  resources: AddResourceOptions[]
+): {
+  classicModelOptions: Array<AddModelOptions<ClassicDataSourceType>>;
+  cadOrPointCloudResources: Array<AddCadResourceOptions | AddPointCloudResourceOptions>;
```

Comment:
It seems to me that this function does two separate things - one is finding which models in the list are CAD/Pointcloud, and then a list of classic identifiers from them. Does it make sense that both happen in the same hook?